### PR TITLE
feat(curriculum): add javascript trivia bot lab to cert

### PR DESCRIPTION
**Pull request by majestic-owl448**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

**Review by jdwilkin4 — COMMENTED** [review]
Here is some early feedback 👍

File: client/i18n/locales/english/intro.json [hunk 1]
```diff
@@ -1921,7 +1921,10 @@
           "You will learn about variables, <code>let</code>, <code>const</code>, <code>console.log</code> and basic string usage."
         ]
       },
-      "zmzg": { "title": "132", "intro": [] },
+      "lab-chat-bot": {
+        "title": "Build a Chat Bot",
```
[the remark this answers is not in the document]
Comment (ilenia-magoni) [reply client/i18n/locales/english/intro.json:1925]: I don't think "JavaScript" is the best, as the language to talk about is not mandated

File: curriculum/challenges/english/25-front-end-development/lab-chat-bot/66ed41f912d0bb1dc62da5dd.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab.
+
+**User Stories:**
+
+1. You should log `"Hello! I'm your coding fun fact guide!"` to the console as a greeting message to the user.
+1. You should create three const variables: `botName`, `botLocation`, and `favoriteLanguage`, that store the bot's name, where it's from, and it's favorite coding language, respectively.
```
[the remark this answers is not in the document]
Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/lab-chat-bot/66ed41f912d0bb1dc62da5dd.md:15]: These are the user stories as they were written and approved, I will make the change

File: curriculum/challenges/english/25-front-end-development/lab-javascript-trivia-bot/66ed41f912d0bb1dc62da5dd.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+# --hints--
+
+You should log `"Hello! I'm your coding fun fact guide!"` to the console.
+
+```js
+assert.equal(output[0], "Hello! I'm your coding fun fact guide!")
+```
+
+You should create a `botName` variable and give it a value.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-javascript-trivia-bot/66ed41f912d0bb1dc62da5dd.md:34]:
```suggestion
You should declare a `botName` variable and give it a value.
```

File: curriculum/challenges/english/25-front-end-development/lab-javascript-trivia-bot/66ed41f912d0bb1dc62da5dd.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+```js
+assert.exists(botName);
+assert.isString(botName);
+```
+
+You should create a `botLocation` variable and give it a value.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-javascript-trivia-bot/66ed41f912d0bb1dc62da5dd.md:41]:
```suggestion
You should declare a `botLocation` variable and give it a value.
```

File: curriculum/challenges/english/25-front-end-development/lab-javascript-trivia-bot/66ed41f912d0bb1dc62da5dd.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+```js
+assert.exists(botLocation);
+assert.isString(botLocation);
+```
+
+You should create a  `favoriteLanguage` variable and give it a value.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-javascript-trivia-bot/66ed41f912d0bb1dc62da5dd.md:48]:
```suggestion
You should declare a `favoriteLanguage` variable and give it a value.
```

File: curriculum/challenges/english/25-front-end-development/lab-javascript-trivia-bot/66ed41f912d0bb1dc62da5dd.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+You should log to the console `"My favorite programming language is (favoriteLanguage)."` using concatenation to add the variable to the string.
+
+```js
+const codeWithoutComments = __helpers.removeJSComments(code);
+assert.equal(output[2], `My favorite programming language is ${favoriteLanguage}.`)
+assert.match(codeWithoutComments, /language is ('|")\s*\+\s*favoriteLanguage\s*\+\s*('|")\./);
+```
+
+You should create a new variable `codingFact` with `let`.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-javascript-trivia-bot/66ed41f912d0bb1dc62da5dd.md:71]:
```suggestion
You should use `let` to declare a new variable `codingFact`.
```

File: curriculum/challenges/english/25-front-end-development/lab-javascript-trivia-bot/66ed41f912d0bb1dc62da5dd.md [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+```js
+const codeWithoutComments = __helpers.removeJSComments(code);
+const [first, second, third] = codeWithoutComments.match(/(let )?\s*codingFact\s*=\s*("|').+?\2\s*\+\s*favoriteLanguage/g)
+assert.match(code, /let\s+codingFact/)
+assert.include(first, 'let');
+assert.exists(first);
+```
+
+You should give a value to `codingFact` that includes `favoriteLanguage` using concatenation.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-javascript-trivia-bot/66ed41f912d0bb1dc62da5dd.md:81]:
```suggestion
You should give `codingFact` a value  that includes `favoriteLanguage` using concatenation.
```

**Review by Dario-DC — COMMENTED** [review]
Left a few comments.

File: client/i18n/locales/english/intro.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (ilenia-magoni) [reply client/i18n/locales/english/intro.json]: I would suggest "Trivia Bot", removing "JavaScript". Or "Programming Trivia Bot"?

**Review by jdwilkin4 — APPROVED** [review]
Looks good 👍

**Review by Dario-DC — APPROVED** [review]

**Pull request merged**